Fix PHP 8 warnings [#23]

fembuelita opened the pull request:
<!--
Thanks for contributing !

Please note :
- These comments won't show up when you submit the pull request.
- Please make sure your changes respect the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards).
- In case you introduced a new action or filter hook, please also include [inline documentation](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php).
- Please provide tests, if you can.
-->

This pull request fixes issue #23 .

It will apply the following changes :

* Updates the singleton class to remove the word `final` from private methods
* Sets the `__wakeup` method to public

fembuelita committed:
Update for PHP8.0 compatibility

Private functions cannot be used with final as private is final by definition. Fixes warnings in PHP8

fembuelita committed:
Merge pull request #1 from elliottpost/fix-php8-warnings

Update for PHP8.0 compatibility

elliottpost commented:
Hey @jameelmoses , with PHP 7.4 on it's way out the door (only in security support for ~6 months more), can we get this merged in, please? This is the only plugin we're waiting to be updated on a couple client sites that prevents us from using PHP 8

fembuelita committed:
Change __clone and __wakeup to public

PHP8 also requires these are public. They return void and are marked as final so they may as well be private

Vincent-Noben commented:
Hey @jameelmoses
Can you please get the ball rolling on this.

PHP8 has been with us for some time now, and your plugin has even been pulled from the WordPress plugin repository due to php8 incompatibility. It only takes 15 minutes of your time to accept some pull requests and get the plugin back out there again.

Much appreciated.

jameelmoses commented:
This plugin has been deprecated in favor of [Flexible Content Extended for Advanced Custom Fields](https://github.com/jameelmoses/acf-flexible-content-extended)
